common_types/src/schema.rs

```diff
@@ -595,6 +630,16 @@ impl Schema {
         self.timestamp_index
     }
 
+    /// Whether i-nth column is key column
+    /// TODO: should tag column be key columns?
+    pub fn non_key_column(&self, i: usize) -> bool {
```

Comment:
Maybe what we need is a just a method called `is_key_column` and that works too by reverting the returned boolean when check whether one column is a non-key column.